File: app/src/main/java/com/mikhailsv/lupus/myapplicationjsoup/changeDisplayText.java
package com.mikhailsv.lupus.myapplicationjsoup;

/**
 * Created by Lupus on 09.12.2017.
 */

//Editing string for displaying in app activity
class changeDisplayText {
    public String displayText(String myString1){

        myString1 = myString1.replaceAll("\\d","").replaceAll("\\(.*?\\)","").replaceAll(",","")
        .replaceAll("\\.", "");
        return myString1;
    }


    //Editing string for to be used in search query
    public String searchText(String myString2){
        //myString2 = myString2.split(",")[0];
        myString2 = myString2.replaceAll(",","");
        myString2 = myString2.replaceAll("\\s","_");
        myString2 = myString2.replaceAll("\\d","");
        myString2 = myString2.replaceAll("\\(","");
        myString2 = myString2.replaceAll("\\)","");
        myString2 = myString2.replaceAll("-","");
        myString2 = myString2.replaceAll("\\.","");

        return myString2;

    }

}

File: app/src/main/java/com/mikhailsv/lupus/myapplicationjsoup/ChangeDisplayText.kt
package com.mikhailsv.lupus.myapplicationjsoup


// Editing text to be displayed
fun displayText(myString1: String): String = myString1.replace("\\d".toRegex(), "").replace("\\(.*?\\)".toRegex(), "").replace(" ,".toRegex(), ",")
            .replace("\\.".toRegex(), "")


//Editing string for to be used in DB search query
fun searchText(myString2: String): String = myString2.replace(",".toRegex(), "").replace("\\s".toRegex(), "_").replace("\\d".toRegex(), "")
            .replace("\\(".toRegex(), "").replace("\\)".toRegex(), "").replace("-".toRegex(), "")
            .replace("\\.".toRegex(), "")

// Text formatted for web search engine
fun searchTextWeb(myString3: String): String = myString3.replace(",".toRegex(), "").replace("\\s".toRegex(), "+").replace("\\d".toRegex(), "")
        .replace("\\(".toRegex(), "").replace("\\)".toRegex(), "").replace("-".toRegex(), "")
        .replace("\\.".toRegex(), "")
